flow/core/kernel/base_energy.py

```diff
@@ -0,0 +1,32 @@
+import dill as pickle
+import boto3
+import botocore
+import math
+import random
+import statistics
+import numpy as np
+from scipy.interpolate import interp1d
+from collections import namedtuple
+from abc import ABCMeta, abstractmethod
+
+
+class BaseEnergyModel(metaclass=ABCMeta):
+    """Base energy model class.
+
+    Calculate the instantaneous power consumption of a vehicle in
+    the network.  It returns the power in Watts regardless of the 
+    vehicle type: whether EV or Combustion Engine, Toyota Prius or Tacoma
+    or non-Toyota vehicles. Non-Toyota vehicles are set by default 
+    to be an averaged-size vehicle.
+    """
+
+    def __init__(self,kernel):
+        self.vehicle = vehicle
+
+    @abstractmethod
+    def get_instantaneous_power(self, veh_id, model_param, grade):
```

Comment:
i've renamed this function, so change accordingly below.

since you've instantiated `self.vehicle` i don't think you need `veh_id` anymore. in fact, you should probably also instantiate the `env` in the `__init__` so that you don't need to supply the grade or other environment variables here (like speed, acceleration).